### ISSUE TITLE
star at import not honored

### ISSUE THREAD

**Issue by TurboGit** [darktable-org/darktable#7692]
**Describe the bug**
The number of star at import is not honored, all images get 0 star.

**To Reproduce**
1. Go to 'lighttable'
2. Click on 'preference' on import module
3. Ensure 1 star is set for new import
4. Import some picture
5. See that 0 star is set on imported images

**Expected behavior**
The number of star for new imported images should be set properly

**Platform (please complete the following information):**
 - Darktable Version: master
 - OS: GNU/Linux 

**Additional context**
I suspect an issue due to recent rewrite of the settings from general pref to import dialog.
I properly have `ui_last/import_initial_rating=1`

**Comment by TurboGit** [comment]
BTW, this happens only when importing new images from a card.

**Comment by MStraeten** [comment]
on osx this happens also if importing new images from internal disks.

**Comment by chhil** [comment]
@TurboGit 
https://github.com/darktable-org/darktable/issues/6904

**Comment by phweyland** [comment]
On my side whatever the import, that works properly.
The only point, is when I import an image which has an xmp, that's the rating from xmp which is retained.

**Comment by TurboGit** [comment]
> The only point, is when I import an image which has an xmp, that's the rating from xmp which is retained.

Which is expected.

Again I see that I was not clear and so I also note this here: This happens only for the import from an external card reader.

**Comment by phweyland** [comment]
Can you check if you have Exif.Image.Rating or Exif.Image.RatingPercent defined in your raw file ?

**Comment by TurboGit** [comment]
Could be the issue indeed:

```
$ exiftool -P 20210105_0001.NEF | grep Rati
Rating                          : 0
```

**Comment by TurboGit** [comment]
But dt should overwrite the value as we have asked for N star at import, no ?

**Comment by MStraeten** [comment]
my testcase on osx:
copy images from sd card  into a directory on internal disk --> the directory doesnt contain any xmp files.
So when importing it in darktable there's no xmp file where a rating can be taken from...
but 
```
exiftool -P Pictures/Import_20210101/20210101-IMG_0309.CR2 | grep Rating
Rating                          : 0
```

**Comment by phweyland** [comment]
> But dt should overwrite the value as we have asked for N star at import, no ?

That's another story.
If we apply always the import rating, it's useless to look at the exif one, or, from another standpoint, it is impossible to use the exif one.
So, as was coded the read exif, the current behavior is correct.
But we could want to force the import rate as well. Could be done with a special value of import rating or a specific force import rating.

**Comment by MStraeten** [comment]
yes, it get 0 even when setting the preference to 2 stars. But thats not my expectation why i set it to 2. 
In my camera the rating is set to "off"

**Comment by phweyland** [comment]
> yes, it get 0 even when setting the preference to 2 stars. But thats not my expectation why i set it to 2.
> In my camera the rating is set to "off"

I've fixed an issue (#6752) some time ago, where a user was claiming that "0" is not "no rating" but "rating 0".
Your camera seems to behave the other way, using the value "0" as "no rating".

I'll try to find out if there an exif standard/norm about this...

**Comment by TurboGit** [comment]
> Without xmp file, you should get always rating 0, whatever "import setting" is. Is that correct ?

Right.

> But we could want to force the import rate as well. Could be done with a special value of import rating or a specific force import rating flag.

I came to the same conclusion. Maybe a checkbox to ask to set the rating ? Or if set to 0 in dt use the exif one ?

**Comment by phweyland** [comment]
It seems that neither of the EXIF Rating tags are standard EXIF. We'll have to find our way ...

> Maybe a checkbox to ask to set the rating ? 

Can be a flexible solution. My preferred one I think.

> Or if set to 0 in dt use the exif one ?

But in exif, the user may have set 1 or 2...

Another option is not to read exif (as it is not standard0), it seems to come from windows and lightroom... A bit harsh.

**Comment by ptilopteri** [comment]
maybe don't use the rating info from the exif data, but then
you will have noise from users who's cameras allow setting
a rating from within their camera

maybe best, make it an import setting as now one can make preferences
there and/or presets

solves both "problems", but now you have to continually
advise people of that availability.  it doesn't end, rtfm!

**Comment by TurboGit** [comment]
> Another option is not to read exif rating (as it is not standard), it seems to come from windows and lightroom... A bit harsh.

I don't think that it would be a good long term solution. If a camera let user's add a rating better to use it if possible even if not standard. So yes, I would go with a pref on dt to set or not the rating.

**Comment by TurboGit** [comment]
@ptilopteri : yes, I think we are converging to a pref.

**Comment by TurboGit** [comment]
@phweyland : can I let you propose a PR for this ?

**Comment by phweyland** [comment]
Something like "do not import exif rating" ?
Because "force import rating" maybe misleading for the xmp one, which has the priority.

**Comment by ptilopteri** [comment]
leaves the question of what the rating will be, what one configs or "0" or ... 

default to "0"
follow configuration # 
or set a prefix to ???

**Comment by phweyland** [comment]
> leaves the question of what the rating will be, what one configs or "0" or ...
> 
> default to "0"
> follow configuration #
> or set a prefix to ???

??

The issue above is that the `initial import rating` is unconsidered when exists an `exif rating`.
With the proposal "do not import exif rating" flag (FALSE per default to avoid behavior change):
- if TRUE, the `initial import rating` is always used (including 0)
- else if no `exif rating`, `initial import rating` is used
- otherwise `exif rating` is used

The unique default option is the default `initial import rating`.

`xmp rating`, if any, overrides the previous one.

Does that make sense ?

**Comment by TurboGit** [comment]
> Something like "do not import exif rating" ?

Yes, would be good indeed and this convey the fact that it is only for exif (from card/camera) import) as we want to keep the rating in the xmp.

**Comment by zisoft** [comment]
Reading through the comments on this issue I assume that it has been resolved.

Unfortunately, it does not work with my Canon RAW files (CR2). Rating is off in my camera settings.

In the import options I set "Ignore EXIF rating" to TRUE and set "Initial rating" to 1:

![image](https://user-images.githubusercontent.com/4083281/212541060-e6b57559-a958-4023-bd4e-42cc3996b945.png)

All imported images have a "0" star rating.

    exiftool -P test_img.CR2 | grep Rating    
    Rating                          : 0

    exiv2 -PEXI test_img.CR2| grep Rating
    Exif.Photo.ISOSpeedRatings                   Short       1  800
    Xmp.xmp.Rating                               XmpText     1  0


I think if "Ignore EXIF rating" is set it should override all?
